main.py

```diff
@@ -0,0 +1,160 @@
+# TODO
+# - Add error handling
+# - Externalize configuration
+# - Add support for SSL
+
+from flask import Flask, request, jsonify
+from markupsafe import Markup
+from jwcrypto import jwt, jwk
+
+from irods.session import iRODSSession
+from irods.models import Model
+from irods.column import Column, Integer, String
+from irods.ticket import Ticket
+from irods.api_number import api_number
+from irods.message import iRODSMessage, TicketAdminRequest
+
+import base64
+import json
+
+app = Flask(__name__)
+
+IRODS_TICKET_BOOTH_SHARED_SECRET = 'THIS IS A SHARED SECRET!'
+
+# TODO Port this to the PRC.
+#
+# The PRC does not have GenQuery support for tickets. Defining the following class
+# enables this, but only for the columns of interest.
+#
+# See the following for additional details:
+#   - https://github.com/irods/python-irodsclient/blob/master/irods/models.py
+#   - https://github.com/irods/python-irodsclient/blob/master/irods/ticket.py
+#   - https://groups.google.com/g/iROD-Chat/c/gVfxH1Nn3Xc/m/EB1YchTOAAAJ
+#   - https://github.com/irods/python-irodsclient/commit/efd1c7495f454242dce81f03e598741cfabd955b#diff-371634f3c7d0926eb7a5118ddef62e79ee9ef2b3e12f893cc49cbe4033e7d270
+class GenQueryTicket(Model):
+    id              = Column(Integer, 'TICKET_ID',        2200)
+    string          = Column(String,  'TICKET_STRING',    2201)
+    type            = Column(String,  'TICKET_TYPE',      2202)
+    collection_name = Column(String,  'TICKET_COLL_NAME', 2228)
+
+def get_username_and_password(auth_header):
+    # Remove the "Basic " prefix.
+    auth_header = request.headers.get('authorization')[6:]
+    app.logger.info('auth header = [%s]', auth_header)
+
+    username, password = base64.b64decode(auth_header).decode('utf-8').split(':')
+    app.logger.info('username = [%s], password = [%s]', username, password)
+
+    return username, password
+
+def make_irods_credentials_dict(username, password):
+    return {
+        'host': 'localhost',
+        'port': 1247,
+        'user': username,
+        'password': password,
+        'zone': 'tempZone'
```

Comment:
port and zone should be parameters as well?

or at least... should be defined in the deployment configuration for this app, so it's locked to a single zone.   oh, you said 'externalize configuration' in the PR notes...

Reply:
done.

see config.py and `app.config.from_envvar`.